src/notifications/datamodel/models.py

```diff
@@ -29,6 +35,15 @@ class Meta:
     def __str__(self) -> str:
         return f"{self.naam}"
 
+    def match_filter_names(self, obj_filters):
+        if not (self.filters and obj_filters):
```

Comment:
wouldn't the simplest check be: `return self.filters == obj_filters`?

* If there are no kanaal filters and obj filters are povided -> no match
* If there are filters but the order or amount is different -> no match
* If there are filters but no object filters -> no match

It only matches if the filters have the same names, amount and order

Reply:
The logic behind this function is to include an option when the number of filters could differ.
For example: If kanaal has filters ['bron', 'zaaktype'] and abonnement has filters ['bron'] this should be considered as a match. 
And as far as I get it no filters there should be a match (i.e. abonnement can subscribe to a kanaal without specified filters)

So I though the option should be following:
- no kanaal filters -> match
- no objet filters -> match 
- kanaal filters = object filters -> match
- kanaal filters is the subset of the first n object filters -> match
- object filters is the subset of the first n kanaal filters -> match
- the rest cases -> no match

Reply:
Sidenote: I suddenly wonder why we use a list of items. If the order is determined by the channel creator and the subscriptions use a filter that has these keys, why do we still need order again? Anyway, discussion for later :)

> For example: If kanaal has filters ['bron', 'zaaktype'] and abonnement has filters ['bron'] this should be considered as a match.

I agree. We implicitly consider the left out filter as binding "*".
This makes it nicer to add additional kenmerken later without hurting existing subscriptions/implementations.

> And as far as I get it no filters there should be a match (i.e. abonnement can subscribe to a kanaal without specified filters)

I agree as well. I don't see the point in having a subscription with no filters resulting in no messages being sent to you. So basically this results in a subscription with binding "#".

> If there are no kanaal filters and obj filters are povided -> no match

I think this will be solved by subscription validation (issue in progress) where you can not take a subscription with filters that don't exist for that channel.

If this leaves room for discussion, I don't think the matching pattern should be the place to solve this, but the subscription validation (like, it should include ALL kenmerken or fail).

Reply:
@annashamray you're right, that flexibility is useful. it also allows to gradually add kenmerken without breaking existing subscriptions, probably.

> I suddenly wonder why we use a list of items. If the order is determined by the channel creator and the subscriptions use a filter that has these keys, why do we still need order again? Anyway, discussion for later :)

the order wasn't determined by the channel creator before, that's why it was needed. We can flatten it and should do so, but not before the demo anymore I'm afraid.

Reply:
Resolving this, it's fine as is I think